Fix all deprecations with: make policy

maage opened the pull request:
I've included deprecate interfaces etc. to commit messages, allow easy verifications.

pp-files should be the same as without this patchset. base.conf is somewhat reordered.

I did fist three patches by hand.

Rest were just 1-1 replacements made by:
```|sh
#!/bin/bash

set -ex

while read -r a b; do

    [ "$a" ] || continue
    [ "$b" ] || continue

    # Ensure we are not handling regexps
    [[ ! "$a" =~ [\\/()] ]]
    [[ ! "$b" =~ [\\/()] ]]

    # Ensure replacement is the same as original sans comments
    atmp="$(mktemp)"
    btmp="$(mktemp)"
    macro-expander -M "$a(zzz_t)" | sed '/^#/d' > "$atmp"
    macro-expander -M "$b(zzz_t)" | sed '/^#/d' > "$btmp"
    cmp -s -- "$atmp" "$btmp" || {
        diff -u -- "$atmp" "$btmp"
        rm -- "$atmp" "$btmp"
        exit 1
    }
    rm -- "$atmp" "$btmp"

    # Just target files where deprecated entry was used.
    # Ensure we do not modify intenface definitions.
    git grep -l -- "$a" | \
        xargs -r perl -pi -e 's/\b(?<!`)\Q'"$a"'\E(?!'"'"')\b/'"$b"'/'

    printf "change deprecated %s -> %s\n\n" "$a" "$b" > "$atmp"
    git grep -l -- "$a" | xargs -r sed -nr "/(define|interface|template)\(\`$a'/,/^'\)/p" >> "$atmp"

    grep -E " $b[( ]" "$atmp"

    git commit -F "$atmp" .

done <<\EOF

auth_read_all_dirs_except_shadow files_list_non_auth_dirs
auth_read_all_files_except_shadow files_read_non_auth_files
auth_read_all_symlinks_except_shadow files_read_non_auth_symlinks

userdom_user_tmpfs_file userdom_user_tmp_file
userdom_user_tmpfs_content userdom_user_tmp_content
userdom_rw_user_tmpfs_files userdom_rw_user_tmp_files
userdom_read_user_tmpfs_files userdom_read_user_tmp_files
userdom_dontaudit_setattr_user_tmpfs userdom_dontaudit_setattr_user_tmp
userdom_manage_tmpfs_role userdom_manage_tmp_role

xserver_manage_xdm_tmp_dirs userdom_manage_user_tmp_dirs
xserver_relabel_xdm_tmp_dirs userdom_relabel_user_tmp_dirs
xserver_dontaudit_xdm_tmp_dirs userdom_dontaudit_setattr_user_tmp
xserver_search_xdm_tmp_dirs userdom_search_user_tmp_dirs

xserver_rw_xdm_tmp_files userdom_rw_user_tmp_files
xserver_manage_xdm_tmp_files userdom_manage_user_tmp_files
xserver_append_xdm_tmp_files userdom_append_user_tmp_files
xserver_read_xdm_tmp_files userdom_read_user_tmp_files
xserver_dontaudit_read_xdm_tmp_files userdom_dontaudit_read_user_tmp_files

xserver_create_xdm_tmp_sockets userdom_create_user_tmp_sockets
xserver_dontaudit_getattr_xdm_tmp_sockets userdom_dontaudit_user_getattr_tmp_sockets
xserver_xdm_tmp_filetrans userdom_user_tmp_filetrans

miscfiles_read_certs miscfiles_read_generic_certs
miscfiles_manage_cert_files miscfiles_manage_generic_cert_files

mmap_files_pattern mmap_exec_files_pattern

automount_exec_config files_exec_etc_files

EOF
```

maage committed:
change deprecated kerberos_keytab_template -> kerberos_read_keytab+kerberos_use

template(`kerberos_keytab_template',`
	refpolicywarn(`$0($*) has been deprecated.')
	kerberos_read_keytab($2)
	kerberos_use($2)
')

maage committed:
remove deprecated mcs_file_read_all/mcs_file_write_all/mcs_socket_write_all_levels

Was deprecated 2013 412c66c9b8 to nothing, I guess safe to change for
now. Most of it was changed then.

interface(`mcs_file_read_all',`
	refpolicywarn(`$0() has been deprecated, please remove mcs_constrained() instead.')
')
interface(`mcs_file_write_all',`
	refpolicywarn(`$0() has been deprecated, please remove mcs_constrained() instead.')
')
interface(`mcs_socket_write_all_levels',`
	refpolicywarn(`$0() has been deprecated, please remove mcs_constrained() instead.')
')

WOnder93 commented:
Thank you for the effort! I have just one comment: I think you should move the new userdom_*() calls out from the xserver `optional_policy()` blocks so that they are together with the other userdom_*() calls (if any). And a few other interface calls could be moved together with the other calls with the same prefix. Granted, it's extra work, but like this it's easier to miss them when browsing the policy files.

maage committed:
policy: init.te: typo: enabled_mls -> enable_mls

maage committed:
policy: java: drop corenet if which need type

maage committed:
policy: java: do not try to use interfaces with attribute that require type

maage committed:
policy: thunderbird: gnome_manage_generic_home_content deletedd some time ago

maage committed:
Was deprecated 2013 c9bd301 and has since done nothing so I guess it is safe to remove

interface(`fs_list_pstorefs',`
	refpolicywarn(`$0($*) has been deprecated.')
')

maage committed:
change deprecated auth_read_all_dirs_except_shadow -> files_list_non_auth_dirs

interface(`auth_read_all_dirs_except_shadow',`
	refpolicywarn(`$0($*) has been deprecated, use files_list_non_auth_dirs() instead.')
	files_list_non_auth_dirs($1)
')

maage committed:
change deprecated auth_read_all_files_except_shadow -> files_read_non_auth_files

interface(`auth_read_all_files_except_shadow',`
	refpolicywarn(`$0($*) has been deprecated, use files_read_non_auth_files() instead.')
	files_read_non_auth_files($1)
')

maage committed:
change deprecated auth_read_all_symlinks_except_shadow -> files_read_non_auth_symlinks

interface(`auth_read_all_symlinks_except_shadow',`
	refpolicywarn(`$0($*) has been deprecated, use files_read_non_auth_symlinks() instead.')
	files_read_non_auth_symlinks($1)
')

maage committed:
change deprecated userdom_user_tmpfs_file -> userdom_user_tmp_file

interface(`userdom_user_tmpfs_file',`
    refpolicywarn(`$0($*) has been deprecated, use userdom_user_tmp_file() instead.')
    userdom_user_tmp_file($1)
')

maage committed:
change deprecated userdom_user_tmpfs_content -> userdom_user_tmp_content

interface(`userdom_user_tmpfs_content',`
    refpolicywarn(`$0($*) has been deprecated, use userdom_user_tmp_content() instead.')
    userdom_user_tmp_content($1)
')

maage committed:
change deprecated userdom_rw_user_tmpfs_files -> userdom_rw_user_tmp_files

interface(`userdom_rw_user_tmpfs_files',`
    refpolicywarn(`$0($*) has been deprecated, use userdom_rw_user_tmp_files() instead.')
    userdom_rw_user_tmp_files($1)
')

maage committed:
change deprecated userdom_read_user_tmpfs_files -> userdom_read_user_tmp_files

interface(`userdom_read_user_tmpfs_files',`
    refpolicywarn(`$0($*) has been deprecated, use userdom_read_user_tmp_files() instead.')
    userdom_read_user_tmp_files($1)
')

maage committed:
change deprecated userdom_dontaudit_setattr_user_tmpfs -> userdom_dontaudit_setattr_user_tmp

interface(`userdom_dontaudit_setattr_user_tmpfs',`
    refpolicywarn(`$0($*) has been deprecated, use userdom_dontaudit_setattr_user_tmp() instead.')
    userdom_dontaudit_setattr_user_tmp($1)
')

maage committed:
change deprecated userdom_manage_tmpfs_role -> userdom_manage_tmp_role

interface(`userdom_manage_tmpfs_role',`
    refpolicywarn(`$0($*) has been deprecated, use userdom_manage_tmp_role() instead.')
    userdom_manage_tmp_role($1,$2)
')

maage committed:
change deprecated xserver_manage_xdm_tmp_dirs -> userdom_manage_user_tmp_dirs

interface(`xserver_manage_xdm_tmp_dirs',`
    refpolicywarn(`$0() has been deprecated, please use userdom_manage_user_tmp_dirs instead.')
    userdom_manage_user_tmp_dirs($1)
')

maage committed:
change deprecated xserver_relabel_xdm_tmp_dirs -> userdom_relabel_user_tmp_dirs

interface(`xserver_relabel_xdm_tmp_dirs',`
    refpolicywarn(`$0() has been deprecated, please use userdom_relabel_user_tmp_dirs instead.')
    userdom_relabel_user_tmp_dirs($1)
')

maage committed:
change deprecated xserver_dontaudit_xdm_tmp_dirs -> userdom_dontaudit_setattr_user_tmp

interface(`xserver_dontaudit_xdm_tmp_dirs',`
    refpolicywarn(`$0() has been deprecated, please use userdom_dontaudit_setattr_user_tmp instead.')
    userdom_dontaudit_setattr_user_tmp($1)
')

maage committed:
change deprecated xserver_search_xdm_tmp_dirs -> userdom_search_user_tmp_dirs

interface(`xserver_search_xdm_tmp_dirs',`
    refpolicywarn(`$0() has been deprecated, please use userdom_search_user_tmp_dirs instead.')
    userdom_search_user_tmp_dirs($1)
')

maage committed:
change deprecated xserver_rw_xdm_tmp_files -> userdom_rw_user_tmp_files

interface(`xserver_rw_xdm_tmp_files',`
    refpolicywarn(`$0() has been deprecated, please use userdom_rw_user_tmp_files instead.')
    userdom_rw_user_tmp_files($1)
')

maage committed:
change deprecated xserver_manage_xdm_tmp_files -> userdom_manage_user_tmp_files

interface(`xserver_manage_xdm_tmp_files',`
    refpolicywarn(`$0() has been deprecated, please use userdom_manage_user_tmp_files instead.')
    userdom_manage_user_tmp_files($1)
')

maage committed:
change deprecated xserver_append_xdm_tmp_files -> userdom_append_user_tmp_files

interface(`xserver_append_xdm_tmp_files',`
    refpolicywarn(`$0() has been deprecated, please use userdom_append_user_tmp_files instead.')
    userdom_append_user_tmp_files($1)
')

maage committed:
change deprecated xserver_read_xdm_tmp_files -> userdom_read_user_tmp_files

interface(`xserver_read_xdm_tmp_files',`
    refpolicywarn(`$0() has been deprecated, please use userdom_read_user_tmp_files instead.')
    userdom_read_user_tmp_files($1)
')

maage committed:
change deprecated xserver_dontaudit_read_xdm_tmp_files -> userdom_dontaudit_read_user_tmp_files

interface(`xserver_dontaudit_read_xdm_tmp_files',`
    refpolicywarn(`$0() has been deprecated, please use userdom_dontaudit_read_user_tmp_files instead.')
    userdom_dontaudit_read_user_tmp_files($1)
')

maage committed:
change deprecated xserver_create_xdm_tmp_sockets -> userdom_create_user_tmp_sockets

interface(`xserver_create_xdm_tmp_sockets',`
    refpolicywarn(`$0() has been deprecated, please use userdom_create_user_tmp_sockets instead.')
    userdom_create_user_tmp_sockets($1)
')

maage committed:
change deprecated xserver_dontaudit_getattr_xdm_tmp_sockets -> userdom_dontaudit_user_getattr_tmp_sockets

interface(`xserver_dontaudit_getattr_xdm_tmp_sockets',`
    refpolicywarn(`$0() has been deprecated, please use userdom_dontaudit_user_getattr_tmp_sockets instead.')
    userdom_dontaudit_user_getattr_tmp_sockets($1)
')

maage committed:
change deprecated xserver_xdm_tmp_filetrans -> userdom_user_tmp_filetrans

interface(`xserver_xdm_tmp_filetrans',`
    refpolicywarn(`$0() has been deprecated, please use userdom_user_tmp_filetrans instead.')
    userdom_user_tmp_filetrans($1,$2, $3, $4)
')

maage committed:
change deprecated miscfiles_read_certs -> miscfiles_read_generic_certs

interface(`miscfiles_read_certs',`
	miscfiles_read_generic_certs($1)
	refpolicywarn(`$0() has been deprecated, please use miscfiles_read_generic_certs() instead.')
')

maage committed:
change deprecated miscfiles_manage_cert_files -> miscfiles_manage_generic_cert_files

interface(`miscfiles_manage_cert_files',`
	miscfiles_manage_generic_cert_files($1)
	refpolicywarn(`$0() has been deprecated, please use miscfiles_manage_generic_cert_files() instead.')
')

maage committed:
change deprecated mmap_files_pattern -> mmap_exec_files_pattern

define(`mmap_files_pattern',`
	# deprecated 20171213
	refpolicywarn(`mmap_files_pattern() is deprecated, please use mmap_exec_files_pattern() instead')
	allow $1 $2:dir search_dir_perms;
	allow $1 $3:file mmap_exec_file_perms;
')

maage committed:
change deprecated automount_exec_config -> files_exec_etc_files

interface(`automount_exec_config',`
	refpolicywarn(`$0(): has been deprecated, please use files_exec_etc_files() instead.')
	files_exec_etc_files($1)
')

maage committed:
policy: accountsd: group userdom_ together

maage committed:
policy: games: group userdom_ together

maage committed:
policy: policykit: group userdom_ together

maage committed:
policy: qemu: group userdom_ together

maage committed:
policy: rpc: group userdom_ together

maage committed:
policy: wireshark: group userdom_ together

maage committed:
policy: locallogin: group userdom_ together

maage committed:
policy: selinuxutil: group userdom_ together

maage committed:
policy: systemd: group userdom_ together

maage committed:
policy: xserver: group userdom_ together

maage committed:
policy: userdomain: group userdom_ together

maage committed:
policy: thunderbird: group userdom_ together

maage committed:
policy: evolution: group userdom_ together

maage committed:
policy: mozilla: group userdom_ together

maage committed:
policy: rhgb: group userdom_ together

maage committed:
policy: sandboxX: group userdom_ together

maage committed:
policy: tftp: group userdom_ together

maage committed:
policy: thumb: group userdom_ together

maage committed:
policy: init: group userdom_ together

maage committed:
policy: abrt: drop duplicate entries within module

maage committed:
policy: accountsd: drop duplicate entries within module

maage committed:
policy: apache: drop duplicate entries within module

maage committed:
policy: cinder: drop duplicate entries within module

maage committed:
policy: dirsrv-admin: drop duplicate entries within module

was exactly same as: corecmd_search_bin(dirsrvadmin_t)

maage committed:
policy: drop usage of dbus_all_session_bus_client as it is gone

maage committed:
policy: gpg: drop duplicate entries within module

maage committed:
policy: mozilla: drop duplicate entries within module

maage committed:
policy: pegasus: drop duplicate entries within module

maage committed:
policy: policykit: drop duplicate entries within module

maage committed:
policy: pulseaudio: drop duplicate entries within module

maage committed:
policy: sandboxX: drop duplicate entries within module

maage committed:
policy: thumb: drop duplicate entries within module

maage committed:
policy: kernel: drop duplicate entries within module

This was same as: kernel_rootfs_mountpoint(unlabeled_t)

maage committed:
policy: xserver: drop duplicate entries within module

maage committed:
policy: init: drop duplicate entries within module

maage committed:
policy: locallogin: drop duplicate entries within module

maage committed:
policy: systemd: drop duplicate entries within module

maage commented:
Okay, now there is next try. I tried to group userdom_ macros and as I did that, I noticed those modules had duplicate entries and I added also patches to clean up those too. Dupes are mostly dropped on modules where there was already modifications in this patch set.

I added also init.te enabled_mls -> enable_mls change.

And some gone entries and then some macros which get type as an arg, not attribute.

As there is those modifications, there is differencies in end results at least in CIL format.
